packages/data-stores/src/onboard/reducer.ts

```diff
@@ -294,6 +294,9 @@ const intent: Reducer< string, OnboardAction > = ( state = '', action ) => {
 	if ( action.type === 'SET_INTENT' ) {
 		return action.intent;
 	}
+	if ( action.type === 'RESET_ONBOARD_STORE' && action?.skipFlags?.includes( 'skipIntent' ) ) {
```

Comment:
If reset the onboard store and include the `skipIntent` flag, then we don't want to delete the site_intent state value.